src/lando/main/models/repo.py

```diff
@@ -72,20 +77,33 @@ class Repo(BaseModel):
     product_details_url = models.CharField(blank=True, default="")
     push_bookmark = models.CharField(blank=True, default="")
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self.is_hg_repo:
+            self.hg = HgRepo(self.system_path or self._system_path)
```

Comment:
Rename one of these `system_path` variables to something else, so it's obvious what the difference is.

Reply:
This naming convention for stored vs. calculated properties is somewhat conventional. The main reason this is added here is because `__init__` could be called before a repo instance is actually saved to the database, and thus `repo.system_path` will not be populated yet).